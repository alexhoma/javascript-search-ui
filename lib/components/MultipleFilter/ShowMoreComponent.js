"use strict";
exports.__esModule = true;
var preact_1 = require("preact");
var Template_1 = require("../Template");
/**
 * Show more component
 *
 * Provides two items
 *   -> Show more element
 *   -> Show less element
 */
var ShowMoreComponent = function (_a) {
    var allItemsLength = _a.allItemsLength, currentLimit = _a.currentLimit, handleShowMore = _a.handleShowMore, handleShowLess = _a.handleShowLess, showMoreContainerClassName = _a.showMoreContainerClassName, showMoreTemplate = _a.showMoreTemplate, showLessTemplate = _a.showLessTemplate, dictionary = _a.dictionary;
    return (allItemsLength > currentLimit)
        ? ((0, preact_1.h)("div", { className: "as-showMore " + showMoreContainerClassName, onClick: handleShowMore },
            (0, preact_1.h)(Template_1["default"], { template: showMoreTemplate, className: "as-showMore--more", dictionary: dictionary })))
        : (allItemsLength === currentLimit)
            ? ((0, preact_1.h)("div", { className: "as-showMore " + showMoreContainerClassName, onClick: handleShowLess },
                (0, preact_1.h)(Template_1["default"], { template: showLessTemplate, className: "as-showMore--less", dictionary: dictionary })))
            : null;
};
exports["default"] = ShowMoreComponent;
