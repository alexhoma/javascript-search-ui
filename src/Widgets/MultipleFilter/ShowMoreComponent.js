/**
 * @jsx h
 */
import { h } from 'preact';
import Template from "../Template";

/**
 * Show more component
 *
 * Provides two items
 *   -> Show more element
 *   -> Show less element
 */
const ShowMoreComponent = ({
    allItems,
    currentLimit,
    handleShowMore,
    handleShowLess,
    showMoreContainerClassName,
    showMoreTemplate,
    showLessTemplate
}) => {
    return (allItems.length > currentLimit)
        ? (
            <div className={`asui-showMore ${showMoreContainerClassName}`}
                 onClick={handleShowMore}
            >
                <Template
                    template={showMoreTemplate}
                    className={`asui-showMore--more`}
                />
            </div>
        )
        : (allItems.length === currentLimit)
            ? (
                <div className={`asui-showMore ${showMoreContainerClassName}`}
                     onClick={handleShowLess}
                >
                    <Template
                        template={showLessTemplate}
                        className={`asui-showMore--less`}
                    />
                </div>
            )
        : null
};

export default ShowMoreComponent;