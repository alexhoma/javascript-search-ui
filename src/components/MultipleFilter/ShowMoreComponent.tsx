import { h } from 'preact';
import Template from "../Template";

/**
 * Show more component
 *
 * Provides two items
 *   -> Show more element
 *   -> Show less element
 */
const ShowMoreComponent = ({
    allItemsLength,
    currentLimit,
    handleShowMore,
    handleShowLess,
    showMoreContainerClassName,
    showMoreTemplate,
    showLessTemplate,
    dictionary,
}) => {
    return (allItemsLength > currentLimit)
        ? (
            <div className={`as-showMore ${showMoreContainerClassName}`}
                 onClick={handleShowMore}
            >
                <Template
                    template={showMoreTemplate}
                    className={`as-showMore--more`}
                    dictionary={dictionary}
                />
            </div>
        )
        : (allItemsLength === currentLimit)
            ? (
                <div className={`as-showMore ${showMoreContainerClassName}`}
                     onClick={handleShowLess}
                >
                    <Template
                        template={showLessTemplate}
                        className={`as-showMore--less`}
                        dictionary={dictionary}
                    />
                </div>
            )
        : null;
};

export default ShowMoreComponent;
