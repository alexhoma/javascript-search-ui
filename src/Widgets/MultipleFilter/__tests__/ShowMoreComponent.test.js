/** @jsx h */
import {h} from 'preact';
import {deep} from 'preact-render-spy';
import ShowMoreComponent from './../ShowMoreComponent';

describe('<ShowMoreComponent />', () => {
    let initialProps = {
        handleShowMore: jest.fn(),
        handleShowLess: jest.fn(),
        showMoreContainerClassName: 'custom-container-class',
        showMoreTemplate: '<b>+ Show more</b>',
        showLessTemplate: '<b>+ Show less</b>'
    };

    it('should render Show More button when allItemsLength is bigger than currentLimit', () => {
        const tree = deep(
            <ShowMoreComponent
                {...initialProps}
                allItemsLength={3}
                currentLimit={2}
            />
        );
        expect(tree).toMatchSnapshot();
    });
    it('should render Show Less button when allItemsLength is equal to currentLimit', () => {
        const tree = deep(
            <ShowMoreComponent
                {...initialProps}
                allItemsLength={3}
                currentLimit={3}
            />
        );

        expect(tree).toMatchSnapshot();
    });
    it('should render nothing allItemsLength is smaller than currentLimit', () => {
        const tree = deep(
            <ShowMoreComponent
                {...initialProps}
                allItemsLength={1}
                currentLimit={3}
            />
        );

        expect(tree).toMatchSnapshot();
    })
});