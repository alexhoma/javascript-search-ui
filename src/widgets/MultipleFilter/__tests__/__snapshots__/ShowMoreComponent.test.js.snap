// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ShowMoreComponent /> should render Show Less button when allItemsLength is equal to currentLimit 1`] = `
preact-render-spy (1 nodes)
-------
<div
  class="as-showMore custom-container-class"
  onClick={[Function mockConstructor]}
>
  <div class="as-showMore--less">
    <b>+ Show less</b>
  </div>
</div>

`;

exports[`<ShowMoreComponent /> should render Show More button when allItemsLength is bigger than currentLimit 1`] = `
preact-render-spy (1 nodes)
-------
<div
  class="as-showMore custom-container-class"
  onClick={[Function mockConstructor]}
>
  <div class="as-showMore--more">
    <b>+ Show more</b>
  </div>
</div>

`;

exports[`<ShowMoreComponent /> should render nothing allItemsLength is smaller than currentLimit 1`] = `
preact-render-spy (1 nodes)
-------
{null}

`;
